cloud/amazon/ec2_vpc_route_table.py

```diff
@@ -0,0 +1,540 @@
+
+    if check_mode and route_table is None:
+        return {'changed': True}
+
+    if route_table is None:
+        try:
+            route_table = vpc_conn.create_route_table(vpc_id)
+        except EC2ResponseError as e:
+            raise AnsibleRouteTableException(
+                'Unable to create route table {0}, error: {1}'
+                .format(route_table_id or resource_tags, e)
+            )
+
+    if propagating_vgw_ids is not None:
+        result = ensure_propagation(vpc_conn, route_table_id,
+                                    propagating_vgw_ids,
+                                    check_mode=check_mode)
+        changed = changed or result['changed']
+
+    if not tags_valid and resource_tags is not None:
+        result = ensure_tags(vpc_conn, route_table.id, resource_tags,
+                             add_only=True, check_mode=check_mode)
+        changed = changed or result['changed']
+
+    if routes is not None:
+        try:
+            result = ensure_routes(vpc_conn, route_table, routes, check_mode)
+            changed = changed or result['changed']
+        except EC2ResponseError as e:
+            raise AnsibleRouteTableException(
+                'Unable to ensure routes for route table {0}, error: {1}'
+                .format(route_table, e)
+            )
+
+    if subnets:
+        associated_subnets = []
+        try:
+            associated_subnets = find_subnets(vpc_conn, vpc_id, subnets)
+        except EC2ResponseError as e:
+            raise AnsibleRouteTableException(
+                'Unable to find subnets for route table {0}, error: {1}'
+                .format(route_table, e)
+            )
+
+        try:
+            result = ensure_subnet_associations(
+                vpc_conn, vpc_id, route_table, associated_subnets, check_mode)
+            changed = changed or result['changed']
+        except EC2ResponseError as e:
+            raise AnsibleRouteTableException(
+                'Unable to associate subnets for route table {0}, error: {1}'
+                .format(route_table, e)
+            )
+
+    return {
+        'changed': changed,
+        'route_table_id': route_table.id,
+    }
+
+
+def main():
+    argument_spec = ec2_argument_spec()
+    argument_spec.update({
+        'vpc_id': {'required': True},
+        'route_table_id': {'required': False},
+        'propagating_vgw_ids': {'type': 'list', 'required': False},
+        'resource_tags': {'type': 'dict', 'required': False},
+        'routes': {'type': 'list', 'required': False},
+        'subnets': {'type': 'list', 'required': False},
+        'state': {'choices': ['present', 'absent'], 'default': 'present'},
+    })
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+    )
+    if not HAS_BOTO:
+        module.fail_json(msg='boto is required for this module')
+
+    ec2_url, aws_access_key, aws_secret_key, region = get_ec2_creds(module)
```

Comment:
Recommend using get_aws_connection_info and then connect_to_aws(boto.vpc,...)

As that handles a lot of things like e.g security tokens, boto profiles etc, so that you don't have to.